src/ScreenManager.cpp

```diff
@@ -365,20 +373,15 @@ bool ScreenManager::IsStackedScreen( const Screen *pScreen ) const
 
 bool ScreenManager::get_input_redirected(PlayerNumber pn)
 {
-	if(pn >= m_input_redirected.size())
-	{
+	if(g_ScreenStack.empty() || pn >= NUM_PLAYERS)
```

Comment:
This should also check `pn >= 0`. Ideally we just use the enum directly instead of treating them as numbers but it's fine for now and we can come back to it.